test-battle.js

```diff
@@ -18,18 +18,28 @@ To run:
 
 */
 
-// potentail cli options
-// assuming flag type cli options need to start with '--', e.g. '--wait'
-// can add other pameters later on
-let cliOptions = { wait: false };
-
-// process cli arguments if any
-for (let i=0; i<process.argv.length; i++){
-    var cParam = process.argv[i].replace('--', '');
-    // if its in the cliOptions list, flip the flag
-    if ( typeof( cliOptions[cParam] ) !== 'undefined' ){
-        cliOptions[cParam] = true;
-    }
+// potentail cli options, and their default values
+let cliOptions = { wait: false,
+                   turns: 15 };
+
+// accepting cli parameters
+var args = require('commander');
+args.description('CLI to test the hero.js code locally')
+    .option('-w, --wait', 'Turn by turn step through of the battle')
+    .option('-t, --turns [int]', 'Specifies how many turns to run', parseInt)
+    .parse(process.argv);
+
+// args validation
+if ( isNaN(args.turns) ){
```

Comment:
I just realized while testing these changes that this condition prevents you from running the battle with default settings. I think this needs to be updated to:
```javascript
if (args.turns && isNaN(args.turn) {
```